### DISCUSSION TITLE
LiteDB is no longer supported?

### DISCUSSION THREAD

**Discussion by Feofilakt** [litedb-org/LiteDB#2107] [General]
Hello! I've noticed on the Insights graphics that the maintainer had stopped commiting. Also there are many opened bugs and unanswered questions. So I'm wondered, is LiteDB no longer supported?

**Comment by SKIDDOW** [comment]
> Hello! I've noticed on the Insights graphics that the maintainer had stopped commiting. Also there are many opened bugs and unanswered questions. So I'm wondered, is LiteDB no longer supported?

I don't think it. Just need to grow more community.

**Comment by bulpet** [comment]
Hi, i want to use your great project, but o need to know what is really status now, supported or not?

**Comment by mbdavid** [comment]
Hellow! After few months working in another comercial project I'm backing to LiteDB. Me and @kcsombrio will back to LiteDB project at January reading and answering issues/PR. He have plans to discover what's needs to be fixed and how can we evolve LiteDB for 2022.

**Reply by luis-fss** [reply]
What a relief to know that, thank you so much for your work.

**Reply by SKIDDOW** [reply]
Interesting ❤️

**Reply by sacdevops** [reply]
I'm following your result, guys 💪🏻 Good Luck

**Reply by Arkanium77** [reply]
LiteDB is absolutely brilliant. Many times it saved me and my friends during prototyping applications!

**Reply by mxd1232** [reply]
Is there any progress on the matter? Will the support/development be continued?

**Comment by dlandi** [comment]
@mbdavid Hello! I don't see any commits since December 21, 2021. 

Can you let us know your status/plans for LiteDB? I am trying to decide if I should use LiteDB for my next project.

**Comment by luizen** [comment]
Yeah, I was asking myself the same question: "Is it ever going to be continued?"
Unfortunately I cannot wait and found this alternative: [RavenDB Embedded](https://ravendb.net/docs/article-page/4.2/csharp/server/embedded). I tried it yesterday and it looks great!

I mean, I don't want to discourage anyone to keep using LiteDB, which is an AMAZING project :) As I understand the team is quite small, and the consequences is that it makes progress a bit slower. That's totally understandable. ;)
Just wanted to give another perspective/alternative for those who cannot wait.

Keep up the great work guys!

**Comment by mbdavid** [comment]
Hi guys, sorry for delay. I miss some github notifications in email.

I created LiteDB 8 year ago, to be a simple "document storage" embedded (in-process) for .NET. Along the year, I was adding more features e creating new versions to be more flexible, usable and adding more features. I love .NET coding, especial for LiteDB (it'a like a son to me 😄) .

But LiteDB still a personal project yet, with no revenue. I tried many times to convert into a professional product, so I can pay for more developer helps to resolve bugfix, write unit tests, write better documentation, make tutorial videos... but do all this stuff alone it's impossible to me. I have a full time developer job and only works on LiteDB on my free time. Convert LiteDB into a commercial-only product are not in my plans.

Current version works great for most cases. But I know there are edge cases or non-tested enviorments that has problems crashing data. Super hard to debug and track errors using my current implementation. There is no dependency injection in current version for better unit tests.

This doesn't mean LiteDB is over. I learn lot of .NET/C# features last few years and still loving coding. I'm re-thinking about all database and file engine to write a simple and safe disk operations. It's will take some months to deliver first pre-alfa version to be tested by developers. I'm working in new repo for now: https://github.com/mbdavid/LiteDB-vNext 

Finally, @luizen, I complete undersant your needs and I agree that comercial database are more robust and testable for a large team. Maybe in your next project you can back here to try LiteDB again ❤️ 

Thanks all for support!

**Reply by luizen** [reply]
Absolutely! :) I'd love to see LiteDB becoming a bad ass product being used by thousands of professional projects in a near future, as popular as SQLite is :) Cheers!

**Reply by Arkanium77** [reply]
New repository - new star. We will closely monitor the progress! 

And thanks again for your work :)

**Comment by tjmoore** [comment]
Is there any scope of bringing this into the .NET Foundation or adoption through Microsoft (keeping open source obviously)?

Commercialisation might clash with Mongo legally, especially if it evolved into more than just a file based database. It should be retained as open source though and preferably no commercial licence to use even if a corporate entity took it on.

Wider adoption is maybe limited being that it's a database specific to .NET, although like Mongo and others you could add wrappers for other (non .NET) languages perhaps.
